test/com/cdd/bao/validator/RemappingCheckerTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * BioAssay Ontology Annotator Tools
+ * 
+ * (c) 2017-2018 Collaborative Drug Discovery Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License 2.0
+ * as published by the Free Software Foundation:
+ * 
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package com.cdd.bao.validator;
+
+import static org.junit.Assert.*;
+
+import java.io.*;
+import java.util.*;
+
+import org.junit.*;
+
+/*
+	Test for logic in RemappingChecker that vets remapping chains
+*/
+
+public class RemappingCheckerTest
+{
+	@Test
+	public void testValidOneLinkChain()
+	{
+		// a => b (valid)
+		Map<String, String> chain = new HashMap<>();
+		chain.put("a", "b");
+		try { RemappingChecker.validateRemappings(chain); }
+		catch (IOException ioe) { assertTrue("No exception should occur here!", false); }
```

Comment:
My convention is that any unit-test for which an exception is expected also declares a throws clause with said exception.  Otherwise, no throws clause is declared.  I agree this leads to more verbose testing logic.  I can see pros/cons either way.